Mpl toolkit fix for zoomed_inset_axes

Tuan333 opened the pull request:
<!--Thank you so much for your PR! To help us review, fill out the form
to the best of your ability.  Please make use of the development guide at
https://matplotlib.org/devdocs/devel/index.html-->

<!--Provide a general summary of your changes in the title above, for
example "Raises ValueError on Non-Numeric Input to set_xlim".  Please avoid
non-descriptive titles such as "Addresses issue #8576".-->

<!--If you are able to do so, please do not create the
PR out of master, but out of a separate branch.  See
https://matplotlib.org/devel/gitwash/development_workflow.html for
instructions.-->

## PR Summary
A small fix for #8589. Test cases provided.
<!--Please provide at least 1-2 sentences describing the pull request in
detail.  Why is this change required?  What problem does it solve?-->

<!--If it fixes an open issue, please link to the issue here.-->

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/whats_new.rst if major new feature
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

<!--We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or the
recommended next step seems overly demanding , or if you would like help in
addressing a reviewer's comments.  And please ping us if you've been waiting
too long to hear back on your PR.-->

Tuan333 commented:
Note: The errors on Appveyor don't seem to reappear in my local machine or an Appveyor built from my forked repo.

QuLogic commented on lib/matplotlib/tests/test_axes.py:
```diff
@@ -5149,3 +5149,21 @@ def test_twinx_knows_limits():
+    fig, ax = plt.subplots()
+    ax.plot([1, 2, 3], [1, 2, 3])
+    ax.axis([1, 3, 1, 3])
+    from mpl_toolkits.axes_grid1.inset_locator import zoomed_inset_axes
```
If it uses `mpl_toolkits`, it doesn't belong in this file. It should go somewhere in `lib/mpl_toolkits/tests`.

Tuan333 replied:
sure. thanks for the pointer

Tuan333 commented:
As with last time, the appveyor build fails, but i can't reproduce. If someone could rebuild to see if it still fails, i'll be thankful

QuLogic commented on lib/mpl_toolkits/tests/test_axes_grid1.py:
```diff
@@ -154,3 +154,21 @@ def get_demo_image():
+    fig, ax = plt.subplots()
+    ax.plot([1, 2, 3], [1, 2, 3])
+    ax.axis([1, 3, 1, 3])
+    from mpl_toolkits.axes_grid1.inset_locator import zoomed_inset_axes
```
This is already imported at the top of the file.

Tuan333 commented:
fixed test case by @QuLogic  comment and squashed commit. If appveyor build still fails (hopefully it won't), someone please rebuild it in my stead, something is quite wrong with my GitHub settings. Thank you.

QuLogic commented:
Probably should have corrected the commit message a bit, though...
